diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/config/i386/mingw32.h gcc-4.7.2/gcc/config/i386/mingw32.h
--- gcc-4.7.2.orig/gcc/config/i386/mingw32.h	2012-08-06 22:34:27.000000000 +0800
+++ gcc-4.7.2/gcc/config/i386/mingw32.h	2012-12-10 22:03:06.392550400 +0800
@@ -121,7 +121,7 @@
 
 /* Include in the mingw32 libraries with libgcc */
 #ifdef ENABLE_SHARED_LIBGCC
-#define SHARED_LIBGCC_SPEC "%{shared-libgcc:-lgcc_s} %{!shared-libgcc:-lgcc_eh}"
+#define SHARED_LIBGCC_SPEC "%{shared-libgcc:-lgcc_s}"
 #else
 #define SHARED_LIBGCC_SPEC /*empty*/
 #endif
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/libgcc/Makefile.in gcc-4.7.2/libgcc/Makefile.in
--- gcc-4.7.2.orig/libgcc/Makefile.in	2012-09-05 20:19:47.000000000 +0800
+++ gcc-4.7.2/libgcc/Makefile.in	2012-12-10 22:03:06.416588600 +0800
@@ -818,7 +818,7 @@
 
 ifeq ($(enable_shared),yes)
 
-libgcc-eh-objects += $(addsuffix $(objext),$(basename $(notdir $(LIB2ADDEHSTATIC))))
+libgcc-objects += $(addsuffix $(objext),$(basename $(notdir $(LIB2ADDEH))))
 libgcc-s-objects += $(addsuffix _s$(objext),$(basename $(notdir $(LIB2ADDEHSHARED))))
 
 iter-items := $(sort $(LIB2ADDEHSTATIC) $(LIB2ADDEHSHARED))
@@ -868,9 +868,8 @@
 libgcc.a: $(libgcc-objects)
 libgcov.a: $(libgcov-objects)
 libunwind.a: $(libunwind-objects)
-libgcc_eh.a: $(libgcc-eh-objects)
 
-libgcc.a libgcov.a libunwind.a libgcc_eh.a:
+libgcc.a libgcov.a libunwind.a:
 	-rm -f $@
 
 	objects="$(objects)";					\
@@ -892,7 +891,7 @@
 endif
 
 ifeq ($(enable_shared),yes)
-all: libgcc_eh.a libgcc_s$(SHLIB_EXT)
+all: libgcc_s$(SHLIB_EXT)
 ifneq ($(LIBUNWIND),)
 all: libunwind$(SHLIB_EXT)
 endif
@@ -1055,10 +1054,6 @@
 install-shared:
 	$(mkinstalldirs) $(DESTDIR)$(inst_libdir)
 
-	$(INSTALL_DATA) libgcc_eh.a $(DESTDIR)$(inst_libdir)/
-	chmod 644 $(DESTDIR)$(inst_libdir)/libgcc_eh.a
-	$(RANLIB) $(DESTDIR)$(inst_libdir)/libgcc_eh.a
-
 	$(subst @multilib_dir@,$(MULTIDIR),$(subst \
 		@shlib_base_name@,libgcc_s,$(subst \
 		@shlib_slibdir_qual@,$(MULTIOSSUBDIR),$(SHLIB_INSTALL))))
