diff --strip-trailing-cr -Nur gcc-4.8.0.orig/libgcc/gthr-posix.h gcc-4.8.0/libgcc/gthr-posix.h
--- gcc-4.8.0.orig/libgcc/gthr-posix.h	2013-02-05 03:06:20.000000000 +0800
+++ gcc-4.8.0/libgcc/gthr-posix.h	2013-03-23 21:45:02.142767800 +0800
@@ -82,7 +82,7 @@
 # define __GTHREAD_COND_INIT_FUNCTION __gthread_cond_init_function
 #endif
 
-#if SUPPORTS_WEAK && GTHREAD_USE_WEAK
+#if SUPPORTS_WEAK && GTHREAD_USE_WEAK && !defined(__MINGW32__)
 # ifndef __gthrw_pragma
 #  define __gthrw_pragma(pragma)
 # endif
