diff --strip-trailing-cr -Nur gcc-4.8.0.orig/gcc/config/i386/mingw32.h gcc-4.8.0/gcc/config/i386/mingw32.h
--- gcc-4.8.0.orig/gcc/config/i386/mingw32.h	2013-03-25 12:51:30.529590200 +0800
+++ gcc-4.8.0/gcc/config/i386/mingw32.h	2013-03-25 12:50:41.414043500 +0800
@@ -78,7 +78,7 @@
 #endif
 
 #undef NATIVE_SYSTEM_HEADER_COMPONENT
-#define NATIVE_SYSTEM_HEADER_COMPONENT "MINGW"
+#undef NATIVE_SYSTEM_HEADER_DIR
 
 #undef CPP_SPEC
 #define CPP_SPEC "%{posix:-D_POSIX_SOURCE} %{mthreads:-D_MT} " \
@@ -152,7 +152,7 @@
 
 /* Override startfile prefix defaults.  */
 #ifndef STANDARD_STARTFILE_PREFIX_1
-#define STANDARD_STARTFILE_PREFIX_1 "/mingw/lib/"
+#define STANDARD_STARTFILE_PREFIX_1 ""
 #endif
 #ifndef STANDARD_STARTFILE_PREFIX_2
 #define STANDARD_STARTFILE_PREFIX_2 ""
diff --strip-trailing-cr -Nur gcc-4.8.0.orig/gcc/incpath.c gcc-4.8.0/gcc/incpath.c
--- gcc-4.8.0.orig/gcc/incpath.c	2013-02-27 15:28:48.000000000 +0800
+++ gcc-4.8.0/gcc/incpath.c	2013-03-25 12:50:41.426051500 +0800
@@ -184,8 +184,7 @@
 	      str = concat (sysroot_no_trailing_dir_separator, p->fname, NULL);
 	      free (sysroot_no_trailing_dir_separator);
 	    }
-	  else if (!p->add_sysroot && relocated
-		   && !filename_ncmp (p->fname, cpp_PREFIX, cpp_PREFIX_len))
+	  if (relocated && !filename_ncmp (p->fname, cpp_PREFIX, cpp_PREFIX_len))
 	    {
  	      static const char *relocated_prefix;
 	      char *ostr;
