diff --strip-trailing-cr -Nur gcc-4.7.2.orig/libiberty/make-relative-prefix.c gcc-4.7.2/libiberty/make-relative-prefix.c
--- gcc-4.7.2.orig/libiberty/make-relative-prefix.c	2012-01-26 22:26:25.000000000 +0800
+++ gcc-4.7.2/libiberty/make-relative-prefix.c	2012-12-10 22:08:11.449665000 +0800
@@ -214,6 +214,22 @@
     }
 }
 
+#ifdef _WIN32
+static int
+win32_is_exe (const char *path)
+{
+  int is_exe = 0;
+  FILE *f = fopen (path, "rb");
+  if (f)
+  {
+    if (fgetc (f) == 'M' && fgetc (f) == 'Z')
+      is_exe = 1;
+    fclose (f);
+  }
+  return is_exe;
+}
+#endif
+
 /* Given three strings PROGNAME, BIN_PREFIX, PREFIX, return a string that gets
    to PREFIX starting with the directory portion of PROGNAME and a relative
    pathname of the difference between BIN_PREFIX and PREFIX.
@@ -281,11 +297,17 @@
 			nstore[endp - startp] = 0;
 		    }
 		  strcat (nstore, progname);
+#ifdef _WIN32
+		  if (win32_is_exe (nstore)
+		      || win32_is_exe (strcat (nstore, HOST_EXECUTABLE_SUFFIX))
+		     )
+#else
 		  if (! access (nstore, X_OK)
 #ifdef HAVE_HOST_EXECUTABLE_SUFFIX
                       || ! access (strcat (nstore, HOST_EXECUTABLE_SUFFIX), X_OK)
 #endif
 		      )
+#endif
 		    {
 #if defined (HAVE_SYS_STAT_H) && defined (S_ISREG)
 		      struct stat st;
