diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/cp/class.c gcc-4.7.2/gcc/cp/class.c
--- gcc-4.7.2.orig/gcc/cp/class.c	2012-07-03 11:30:27.000000000 +0800
+++ gcc-4.7.2/gcc/cp/class.c	2012-12-10 22:19:29.364358700 +0800
@@ -7538,7 +7538,7 @@
   indented = maybe_indent_hierarchy (stream, indent, 0);
   fprintf (stream, "%s (0x%lx) ",
 	   type_as_string (BINFO_TYPE (binfo), TFF_PLAIN_IDENTIFIER),
-	   (unsigned long) binfo);
+	   (uintptr_t) binfo);
   if (binfo != igo)
     {
       fprintf (stream, "alternative-path\n");
@@ -7563,7 +7563,7 @@
       fprintf (stream, " primary-for %s (0x%lx)",
 	       type_as_string (BINFO_TYPE (BINFO_INHERITANCE_CHAIN (binfo)),
 			       TFF_PLAIN_IDENTIFIER),
-	       (unsigned long)BINFO_INHERITANCE_CHAIN (binfo));
+	       (uintptr_t)BINFO_INHERITANCE_CHAIN (binfo));
     }
   if (BINFO_LOST_PRIMARY_P (binfo))
     {
@@ -7696,7 +7696,7 @@
       if (ctor_vtbl_p)
 	{
 	  if (!BINFO_VIRTUAL_P (binfo))
-	    fprintf (stream, " (0x%lx instance)", (unsigned long)binfo);
+	    fprintf (stream, " (0x%lx instance)", (uintptr_t)binfo);
 	  fprintf (stream, " in %s", type_as_string (t, TFF_PLAIN_IDENTIFIER));
 	}
       fprintf (stream, "\n");
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/ggc-common.c gcc-4.7.2/gcc/ggc-common.c
--- gcc-4.7.2.orig/gcc/ggc-common.c	2010-11-30 19:41:24.000000000 +0800
+++ gcc-4.7.2/gcc/ggc-common.c	2012-12-10 22:19:29.377357400 +0800
@@ -308,7 +308,7 @@
   enum gt_types_enum type;
 };
 
-#define POINTER_HASH(x) (hashval_t)((long)x >> 3)
+#define POINTER_HASH(x) (hashval_t)((uintptr_t)x >> 3)
 
 /* Register an object in the hash table.  */
 
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/graphite-dependences.c gcc-4.7.2/gcc/graphite-dependences.c
--- gcc-4.7.2.orig/gcc/graphite-dependences.c	2011-07-22 06:56:29.000000000 +0800
+++ gcc-4.7.2/gcc/graphite-dependences.c	2012-12-10 22:19:29.390358400 +0800
@@ -56,7 +56,7 @@
 {
   const struct poly_ddr *p = (const struct poly_ddr *) pddr;
 
-  return (hashval_t) ((long) PDDR_SOURCE (p) + (long) PDDR_SINK (p));
+  return (hashval_t) ((uintptr_t) PDDR_SOURCE (p) + (uintptr_t) PDDR_SINK (p));
 }
 
 /* Returns true when PDDR has no dependence.  */
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/pointer-set.c gcc-4.7.2/gcc/pointer-set.c
--- gcc-4.7.2.orig/gcc/pointer-set.c	2007-08-16 22:31:21.000000000 +0800
+++ gcc-4.7.2/gcc/pointer-set.c	2012-12-10 22:19:29.407350600 +0800
@@ -64,7 +64,7 @@
 #endif
   const unsigned long shift = HOST_BITS_PER_LONG - logmax;
 
-  return ((A * (unsigned long) p) >> shift) & (max - 1);
+  return ((A * (uintptr_t) p) >> shift) & (max - 1);
 }
 
 /* Allocate an empty pointer set.  */
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/prefix.c gcc-4.7.2/gcc/prefix.c
--- gcc-4.7.2.orig/gcc/prefix.c	2011-06-15 04:11:37.000000000 +0800
+++ gcc-4.7.2/gcc/prefix.c	2012-12-10 22:19:29.421397000 +0800
@@ -157,12 +157,12 @@
     }
 
   size = 32;
-  dst = xmalloc (size);
+  dst = (char *) xmalloc (size);
 
   res = RegQueryValueExA (reg_key, key, 0, &type, (LPBYTE) dst, &size);
   if (res == ERROR_MORE_DATA && type == REG_SZ)
     {
-      dst = xrealloc (dst, size);
+      dst = (char *) xrealloc (dst, size);
       res = RegQueryValueExA (reg_key, key, 0, &type, (LPBYTE) dst, &size);
     }
 
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/print-tree.c gcc-4.7.2/gcc/print-tree.c
--- gcc-4.7.2.orig/gcc/print-tree.c	2011-11-08 19:13:41.000000000 +0800
+++ gcc-4.7.2/gcc/print-tree.c	2012-12-10 22:19:29.438369600 +0800
@@ -255,7 +255,7 @@
   /* Allow this function to be called if the table is not there.  */
   if (table)
     {
-      hash = ((unsigned long) node) % HASH_SIZE;
+      hash = ((uintptr_t) node) % HASH_SIZE;
 
       /* If node is in the table, just mention its address.  */
       for (b = table[hash]; b; b = b->next)
diff --strip-trailing-cr -Nur gcc-4.7.2.orig/gcc/tree-dump.c gcc-4.7.2/gcc/tree-dump.c
--- gcc-4.7.2.orig/gcc/tree-dump.c	2011-06-29 22:29:22.000000000 +0800
+++ gcc-4.7.2/gcc/tree-dump.c	2012-12-10 22:19:29.451362400 +0800
@@ -169,7 +169,7 @@
 dump_pointer (dump_info_p di, const char *field, void *ptr)
 {
   dump_maybe_newline (di);
-  fprintf (di->stream, "%-4s: %-8lx ", field, (unsigned long) ptr);
+  fprintf (di->stream, "%-4s: %-8lx ", field, (uintptr_t) ptr);
   di->column += 15;
 }
 
