diff --strip-trailing-cr -Nur gcc-4.8.0.orig/gcc/Makefile.in gcc-4.8.0/gcc/Makefile.in
--- gcc-4.8.0.orig/gcc/Makefile.in	2013-03-14 17:13:36.000000000 +0800
+++ gcc-4.8.0/gcc/Makefile.in	2013-03-23 22:15:17.477602800 +0800
@@ -1740,6 +1740,13 @@
 native: gengtype$(exeext)
 endif
 
+stage_current_data:= $(shell cat $(toplevel_builddir)/stage_current)
+ifeq ($(stage_current_data),stage1)
+  cp_crt_cmd:=true
+else
+  cp_crt_cmd:=cp -f $(toplevel_builddir)/crt32/crt2.o $(toplevel_builddir)/prev-gcc/32/crt2.o; cp -f $(toplevel_builddir)/crt32/crtbegin.o $(toplevel_builddir)/prev-gcc/32/crtbegin.o; cp -f $(toplevel_builddir)/crt32/crtend.o $(toplevel_builddir)/prev-gcc/32/crtend.o; cp -f $(toplevel_builddir)/crt/crt2.o $(toplevel_builddir)/prev-gcc/crt2.o; cp -f $(toplevel_builddir)/crt/crtbegin.o $(toplevel_builddir)/prev-gcc/crtbegin.o; cp -f $(toplevel_builddir)/crt/crtend.o $(toplevel_builddir)/prev-gcc/crtend.o
+endif
+
 # On the target machine, finish building a cross compiler.
 # This does the things that can't be done on the host machine.
 rest.cross: specs
@@ -3821,6 +3828,7 @@
 ## several C macro definitions, just like version.o
 build/version.o:  version.c version.h \
                   $(REVISION) $(DATESTAMP) $(BASEVER) $(DEVPHASE)
+	-$(cp_crt_cmd)
 	$(COMPILER_FOR_BUILD) -c $(BUILD_COMPILERFLAGS) $(BUILD_CPPFLAGS) \
 	-DBASEVER=$(BASEVER_s) -DDATESTAMP=$(DATESTAMP_s) \
 	-DREVISION=$(REVISION_s) \
