diff --strip-trailing-cr -Nur gcc-4.8.0.orig/libstdc++-v3/config/abi/pre/gnu.ver gcc-4.8.0/libstdc++-v3/config/abi/pre/gnu.ver
--- gcc-4.8.0.orig/libstdc++-v3/config/abi/pre/gnu.ver	2013-02-02 04:44:41.000000000 +0800
+++ gcc-4.8.0/libstdc++-v3/config/abi/pre/gnu.ver	2013-03-25 12:53:24.080153000 +0800
@@ -1536,7 +1536,7 @@
     __cxa_allocate_dependent_exception;
     __cxa_free_dependent_exception;
     __cxa_deleted_virtual;
-
+    _ZTC*;
 } CXXABI_1.3.5;
 
 CXXABI_1.3.7 {
