diff --strip-trailing-cr -Nur gcc-4.8.0.orig/gcc/ada/tracebak.c gcc-4.8.0/gcc/ada/tracebak.c
--- gcc-4.8.0.orig/gcc/ada/tracebak.c	2013-02-14 21:04:10.000000000 +0800
+++ gcc-4.8.0/gcc/ada/tracebak.c	2013-03-23 22:49:26.672865400 +0800
@@ -597,7 +597,7 @@
   cnt = 0;
   while (cnt < size)
     {
-      if (STOP_FRAME (current, top_stack) ||
+      if (STOP_FRAME (current, (FARPROC)top_stack) ||
 	  !VALID_STACK_FRAME(((char *) current->return_address) + PC_ADJUST))
         break;
 
