diff --strip-trailing-cr -Nur gcc-4.8.0.orig/gcc/ada/adaint.c gcc-4.8.0/gcc/ada/adaint.c
--- gcc-4.8.0.orig/gcc/ada/adaint.c	2013-02-06 20:01:20.000000000 +0800
+++ gcc-4.8.0/gcc/ada/adaint.c	2013-03-23 22:51:15.704515200 +0800
@@ -544,7 +544,7 @@
   S2WSC (wdir, dir, GNAT_MAX_PATH_LEN);
   S2WSC (wfile, file, GNAT_MAX_PATH_LEN);
 
-  _stprintf (wfull_path, _T("%s%c%s"), wdir, _T(DIR_SEPARATOR), wfile);
+  _stprintf (wfull_path, GNAT_MAX_PATH_LEN, _T("%s%c%s"), wdir, _T(DIR_SEPARATOR), wfile);
   fd = _topen (wfull_path, O_CREAT | O_EXCL, 0600);
 #else
   char full_path[256];
