diff --strip-trailing-cr -Nur gmp-5.0.5.orig/Makefile.in gmp-5.0.5/Makefile.in
--- gmp-5.0.5.orig/Makefile.in	2012-05-06 19:19:56.000000000 +0800
+++ gmp-5.0.5/Makefile.in	2012-12-11 21:00:08.060671700 +0800
@@ -108,12 +108,12 @@
 libgmp_la_OBJECTS = $(am_libgmp_la_OBJECTS)
 libgmp_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(libgmp_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(libgmp_la_LDFLAGS) $(LDFLAGS) -o $(TARGET_MAYBE64)
 am_libgmpxx_la_OBJECTS = dummy.lo
 libgmpxx_la_OBJECTS = $(am_libgmpxx_la_OBJECTS)
 libgmpxx_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
-	$(CXXFLAGS) $(libgmpxx_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(CXXFLAGS) $(libgmpxx_la_LDFLAGS) $(LDFLAGS) -o $(TARGET_MAYBE64)
 @WANT_CXX_TRUE@am_libgmpxx_la_rpath = -rpath $(libdir)
 am__DEPENDENCIES_2 = $(srcdir)/libmp.sym $(MPBSD_OBJECTS) \
 	$(MPN_OBJECTS) mpz/add$U.lo mpz/gcdext$U.lo mpz/invert$U.lo \
@@ -125,7 +125,7 @@
 libmp_la_OBJECTS = $(am_libmp_la_OBJECTS)
 libmp_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(libmp_la_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	$(LDFLAGS) -o $(TARGET_MAYBE64)
 @WANT_MPBSD_TRUE@am_libmp_la_rpath = -rpath $(libdir)
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp =
@@ -746,12 +746,35 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
+MAYBE64=
+ifeq ($(findstring -m32,$(CFLAGS)),)
+  MAYBE64=_64
+endif
+TARGET_MAYBE64 = $(patsubst %.la,%$(MAYBE64).la,$@)
 libgmp.la: $(libgmp_la_OBJECTS) $(libgmp_la_DEPENDENCIES) 
-	$(libgmp_la_LINK) -rpath $(libdir) $(libgmp_la_OBJECTS) $(libgmp_la_LIBADD) $(LIBS)
+	$(libgmp_la_LINK) -rpath $(libdir) $(libgmp_la_OBJECTS) $(libgmp_la_LIBADD) $(LIBS) \
+	&& rm -f "$@" \
+	&& cat ".libs/$(TARGET_MAYBE64)" | sed 's/\(.*\)_64\(\.dll\)\?\.a/\1\2\.a/' >"$@" \
+	&& cp -p "$@" ".libs/$@" \
+	&& cp -p "$@" ".libs/$@i" \
+	&& mv -f "$(patsubst %.la,%.dll.a,.libs/$(TARGET_MAYBE64))" tmp_gmp \
+	&& mv -f tmp_gmp "$(patsubst %.la,%.dll.a,.libs/$@)"
 libgmpxx.la: $(libgmpxx_la_OBJECTS) $(libgmpxx_la_DEPENDENCIES) 
-	$(libgmpxx_la_LINK) $(am_libgmpxx_la_rpath) $(libgmpxx_la_OBJECTS) $(libgmpxx_la_LIBADD) $(LIBS)
+	$(libgmpxx_la_LINK) $(am_libgmpxx_la_rpath) $(libgmpxx_la_OBJECTS) $(libgmpxx_la_LIBADD) $(LIBS) \
+	&& rm -f "$@" \
+	&& cat ".libs/$(TARGET_MAYBE64)" | sed 's/\(.*\)_64\(\.dll\)\?\.a/\1\2\.a/' >"$@" \
+	&& cp -p "$@" ".libs/$@" \
+	&& cp -p "$@" ".libs/$@i" \
+	&& mv -f "$(patsubst %.la,%.dll.a,.libs/$(TARGET_MAYBE64))" tmp_gmpxx \
+	&& mv -f tmp_gmpxx "$(patsubst %.la,%.dll.a,.libs/$@)"
 libmp.la: $(libmp_la_OBJECTS) $(libmp_la_DEPENDENCIES) 
-	$(libmp_la_LINK) $(am_libmp_la_rpath) $(libmp_la_OBJECTS) $(libmp_la_LIBADD) $(LIBS)
+	$(libmp_la_LINK) $(am_libmp_la_rpath) $(libmp_la_OBJECTS) $(libmp_la_LIBADD) $(LIBS) \
+	&& rm -f "$@" \
+	&& cat ".libs/$(TARGET_MAYBE64)" | sed 's/\(.*\)_64\(\.dll\)\?\.a/\1\2\.a/' >"$@" \
+	&& cp -p "$@" ".libs/$@" \
+	&& cp -p "$@" ".libs/$@i" \
+	&& mv -f "$(patsubst %.la,%.dll.a,.libs/$(TARGET_MAYBE64))" tmp_mp \
+	&& mv -f tmp_mp "$(patsubst %.la,%.dll.a,.libs/$@)"
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
